rsa/parallel.py

```diff
@@ -84,7 +84,44 @@ def getprime(nbits, poolsize):
     return result
 
 
-__all__ = ['getprime']
+def exec_parallel_curry(function,poolsize):
+    """returns a multiprocess version of the supplied function for the given poolsize"""
+    def retfunc(*args,**kwargs):
+        return exec_parallel(poolsize,function,args,kwargs)
+    return retfunc
+
+def _do_work(pipe,function,args,kwargs):
+    result=function(*args,**kwargs)
+    pipe.send(result)
+
+def exec_parallel(poolsize,function,args,kwargs):
+    """
+    carries out a function in multiple processes. Returns the first return value
+    to be produced by any process
```

Comment:
What happens to the other processes and their results? This should be described well.